var/spack/repos/builtin/packages/py-arm-pyart/StringIO.patch

```diff
@@ -0,0 +1,30 @@
+diff -ru a/pyart/testing/data/make_small_mdv_ppi.py b/pyart/testing/data/make_small_mdv_ppi.py
+--- a/pyart/testing/data/make_small_mdv_ppi.py	2022-07-19 13:06:08.000000000 -0500
++++ b/pyart/testing/data/make_small_mdv_ppi.py	2022-09-10 14:26:11.021976219 -0500
+@@ -6,7 +6,10 @@
+ """
+ 
+ import struct
+-import StringIO
++try:
++    from StringIO import StringIO
```

Comment:
Shouldn't this be `import StringIO`? This seems to be different than the original. Again, I wouldn't worry too much about Python 2 support, we're removing it: https://github.com/spack/spack/discussions/31824

Reply:
There were other python files in there that already had the `try:` conditional. I felt it was best to be consistent with what else is in the `pyart` code.

Reply:
That's not how Python works. Later in this file, they run:
```python
fileobj = StringIO.StringIO()
```
This won't work anymore since you're using `StringIO.StringIO.StringIO()` now.

Anyway, based on the directory names, I'm guessing these files are only used to run the unit tests? I wouldn't bother patching those since we aren't testing anything.

Reply:
Those files are used for user testing, not the unit tests. The unit tests are in a different directory and not installed, although they also use `StringIO`. 

It would not be `StringIO.StringIO.StringIO()`, but `StringIO()`. Apparently, this file could only work with python-2. The original was set to import the `StringIO` module and then later use the `StringIO` function in that, so `StringIO.StringIO` in the assignment. I did not catch the later usage call. That will also need to be changed, because what I put in, based on other code, would require the call to be `StringIO()`. In any event, 
```
fileobj = StringIO.StringIO()
```
would not work in either case as that would need to be
```
fileobj = StringIO()
```
when importing with either
```
from io import StringIO
```
or
```
from StringIO import StringIO
```
I did not catch it but that assignment is setup such that it would only work for python-2, which can import `StringIO` as a module. For python-3, `StringIO` comes from the `io` module so `StringIO.StringIO` could never work.

Given all of that, I have to wonder how much the "testing" files get used. So, I can fix the assignment statement or just drop this patch altogether.

Reply:
I pushed a commit to fix the `fileobj` call.